src/Runner.Worker/Handlers/CompositeActionHandler.cs

```diff
@@ -45,13 +46,17 @@ public async Task RunAsync(ActionRunStage stage)
             else if (stage == ActionRunStage.Post)
             {
                 ArgUtil.NotNull(Data.PostSteps, nameof(Data.PostSteps));
-                steps = Data.PostSteps.ToList();
+                steps = new List<Pipelines.ActionStep>();
                 // Only register post steps for steps that actually ran
-                foreach (var step in steps)
+                foreach (var step in Data.PostSteps.ToList())
                 {
-                    if (!ExecutionContext.Root.EmbeddedStepsWithPostRegistered.Contains(step.Id))
+                    if (ExecutionContext.Root.EmbeddedStepsWithPostRegistered.Contains(step.Id))
+                    {
+                        steps.Add(step);
+                    }
+                    else
                     {
-                        steps.Remove(step);
+                        Trace.Info($"Skipping executing post step id: {step.Id}, name: ${step.DisplayName}");
```

Comment:
remind me, what is `Trace`? I thought we used either ExecutionContext or HostContext for tracing. Is `Trace` the HostContext trace writer?

Reply:
Also, should we write this to the execution context instead (i.e. user logs)? I don't know the correct answer, but posing the question.

Reply:
This goes into the worker diag log, which mirrors how we do this in other places. I'll think a bit about a message for the ExecutionContext, might be worth adding in addition.